src/main/scala/org/scalacheck/Shrink.scala

```diff
@@ -14,59 +14,60 @@ import language.higherKinds
 import util.Buildable
 import util.SerializableCanBuildFroms._
 import scala.concurrent.duration.{Duration, FiniteDuration}
+import ScalaVersionSpecific._
 
 sealed abstract class Shrink[T] extends Serializable {
-  def shrink(x: T): Stream[T]
+  def shrink(x: T): LazyList[T]
```

Comment:
So the signature for `Shrink` changes here, so existing 1.14.0 tests (or libraries) that defined shrinkers would no longer compile.

Reply:
The signature is actually the same on 2.10-2.12, because `LazyList` is an alias for `Stream` there: https://github.com/rickynils/scalacheck/blob/543b4056bffdb684faec49e8afb09aee4c24eb2f/src/main/scala-2.13-/org/scalacheck/ScalaVersionSpecific.scala#L19-L20

Reply:
Here's an example shrinker:

```scala
object CustomShrink {
  implicit val dumbIntShrinker: Shrink[Int] = Shrink {
    case _ => Stream.empty
  }
}
```

Publishing this branch as 1.14.0-543b405, the example compiles on 2.10, 2.11, and 2.12, but for 2.13.0-M4:

```
[error] src/main/scala/CustomShrink.scala:5:22: polymorphic expression cannot be instantiated to expected type;
[error]  found   : [A]scala.collection.immutable.Stream[A]
[error]  required: LazyList[Int]
[error]     case _ => Stream.empty
```

Reply:
Thank you for pointing this out.

Do poeple actually write shrinkers "ahead of time", at the same time when writing `Arbitrary` instances, and put them into the code base?

We could make the `LazyList` aliases public on 2.10-2.12, then poeople could use these. But I think this gets even more confusing. Already having the `LazyList` alias in the signatures on 2.10-2.12 is a bit problematic.

We could keep using `Stream`, but that would lead to deprecation warnings on 2.13.

Reply:
Yeah, writing one's own `Shrink` is documented.

Reply:
Aren't we already emitting warnings with `Traversable`?

Reply:
There are a number of uses of `Traversable` in the API, but I think they should not lead to deprecation warnings for the users. In 2.13, `Traversable` is a (deprecated) alias to `Iterable`.

So let's say scalacheck defines a method

```
scala> def foo: Traversable[Int] = List(1,2,3)
                ^
       warning: type Traversable in package scala is deprecated (since 2.13.0): Use Iterable instead of Traversable
foo: Traversable[Int]
```

When compiling scalacheck on 2.13, this will give a deprecation warning. But for users it won't:

```
scala> foo
res0: Traversable[Int] = List(1, 2, 3)

scala> foo.map(_ + 1)
res1: Iterable[Int] = List(2, 3, 4)

scala> foo: Iterable[Int]
res2: Iterable[Int] = List(1, 2, 3)
```

Reply:
Ah, ok.  I see the distinction.  Thanks for explaining.

For now, preserve `Stream` in `Shrink`, and then migrate to `LazyList` in 1.15.0?